### PR TITLE
MAINT: Cleanup Python 2 references

### PR DESCRIPTION
**Pull request by sethtroisi**
#### References to other Issues or PRs

Work on #17943

#### Release Notes

NO ENTRY

### CODE REVIEW

File: sympy/core/cache.py [hunk 1]
```diff
@@ -89,8 +89,6 @@ def __cacheit(maxsize):
         def func_wrapper(func):
             cfunc = lru_cache(maxsize, typed=True)(func)
 
-            # wraps here does not propagate all the necessary info
-            # for py2.7, use update_wrapper below
             def wrapper(*args, **kwargs):
                 try:
```
Comment (oscarbenjamin) [inline sympy/core/cache.py]: Can we just use wraps instead?

**Comment by oscarbenjamin** [comment]
I left one question but otherwise this looks good.

File: sympy/core/cache.py (hunk 1 above)
Comment (sethtroisi) [reply sympy/core/cache.py]: I tried with 
```
        def func_wrapper(func):
            from .decorators import wraps

            cfunc = lru_cache(maxsize, typed=True)(func)

            @wraps(func)
            def wrapper(*args, **kwargs):
                try:
                    retval = cfunc(*args, **kwargs)
                except TypeError:
                    retval = func(*args, **kwargs)
                return retval

            CACHE.append(wrapper)
            return wrapper
```

but this causes a test to fail

```
_____________ sympy/core/tests/test_parameters.py:test_add _____________
Traceback (most recent call last):
  File "/home/eights/Projects/sympy/sympy/core/tests/test_parameters.py", line 29, in test_add
    assert (x + x).args == (2, x)
AssertionError
```

I don't understand this code well enough to understand why this broke, but if you have suggestions I'm happy to poke at it more

Comment (oscarbenjamin) [reply sympy/core/cache.py]: I can't reproduce that error

Comment (sethtroisi) [reply sympy/core/cache.py]: I tested again and I can reproduce locally and in a clean environment.
https://colab.research.google.com/drive/1a54htpAbumrZ1ABNJvankqyl7RTnF4Up

I'm not sure why colab doesn't print output like my local python session does.

I'm happy to either
Revert the changes in just this file and we can submit the other changes
or we can keep debugging until we understand what part of this breaks this test and commit the clearer code here.

Comment (oscarbenjamin) [reply sympy/core/cache.py]: Does it only fail if you run all the tests? I just tried the code from that particular test and from running all the core tests and both passed...

Comment (sethtroisi) [reply sympy/core/cache.py]: Figure it out. I had removed the lines that assign the cfunc cache_info to the wrapper.

I have updated the PR now and everything passes.

**Comment by oscarbenjamin** [comment]
Looks good. Thanks

**Comment by sethtroisi** [comment]
Thanks for considering and advising on how to improve.

**Pull request merged**